mobile/native/src/ln_dlc/mod.rs

```diff
@@ -167,6 +167,42 @@ pub fn run(data_dir: String) -> Result<()> {
             }
         });
 
+        tokio::spawn({
+            let node = node.clone();
+            async move {
+                loop {
+                    tokio::time::sleep(PROCESS_TRADE_REQUESTS_INTERVAL).await;
+
+                    let coordinator_pubkey = config::get_coordinator_info().pubkey;
+                    tracing::debug!(%coordinator_pubkey, "Checking for DLC offers");
+
+                    let sub_channel = match node.get_sub_channel_offer(&coordinator_pubkey) {
+                        Ok(Some(sub_channel)) => sub_channel,
+                        Ok(None) => {
+                            tracing::debug!(%coordinator_pubkey, "No DLC channel offers found");
+                            continue;
+                        },
+                        Err(e) => {
+                            tracing::error!(peer = %coordinator_pubkey.to_string(), "Unable to retrieve DLC channel offer: {e:#}");
+                            continue;
+                        }
+                    };
+
+                    tracing::info!(%coordinator_pubkey, "Found DLC channel offer");
+
+                    let channel_id = sub_channel.channel_id;
+
+                    // todo: the app should validate if the offered dlc channel matches it's submitted order.
```

Comment:
❓How do we associate incoming DLC requests with orders? 

In the previous flow the client was in charge of triggering the `trade` based on the orderbook reporting a match. That match was for an order, and thus the client could identify the order.
The client then triggerd the execution, and only allowed one trade execution at the time - so if multiple match events from the orderbook would have come in, the client was in charge of scheduling the execution.

Now, the coordinator is in charge of scheduling the execution. If the client submitted multiple order to the orderbook, how would the client know which order is being executed here and update the order state accordingly?

By adapting this, the order / position lifecycle updates were removed, I think we should add them again so we can find an answer to the question above :)

Possibilities:

- The client asks the coordinator what is being executed
- The coordinator communicates the order id that is being executed to the client
- ~~The client asks the orderbook what was matched~~ -> I think this does not work, because the orderbook does not execute, and several oders could have been matched, so we run into the same problem

My conclusion is that the information what order is in execution has to come from the coordinator. We can easily validate that the trade parameters that the coordinator proposes match the order in question, so I think this is fine.

Reply:
I still think that the orderbook should notify the trader as well and tell them: 
_hey, your order X,Y,Z have been matched, expect a DLC channel of size Q coming from coordinator c_.

Reply:
> I still think that the orderbook should notify the trader as well and tell them:
> hey, your order X,Y,Z have been matched, expect a DLC channel of size Q coming from coordinator c.

We can do that, but I think we additionally need a way to get information from the coordinator about which execution (i.e. DLC) is for which order. I don't think the orderbook telling us is enough - especially for a maker that potentially has several takers trading with him at the same time. We have to be able to clearly associate one order with one DLC, and I don't think the orderbook has this information in this flow.